src/Adapter/MailTemplate/MailPartialTemplateRenderer.php

```diff
@@ -61,6 +61,12 @@ public function __construct(Smarty $smarty)
      */
     public function render($partialTemplateName, LanguageInterface $language, array $variables = [], $cleanComments = false)
     {
+        Hook::exec('actionMailPartialTemplateRendererVariables', [
```

Comment:
Could you rather use the `HookManager` in constructor ?

Reply:
Hi @Progi1984 
Can you elaborate?

Reply:
We don't use legacy classes in code. We try to use injected dependencies.

Take for example `src/Adapter/Presenter/Product/ProductPresenter.php` which uses HookManager.

Reply:
Hi See that it instantiated the HookManager as a property, but it still calls just `Hook::exec()`:
https://github.com/PrestaShop/PrestaShop/blob/1.7.8.x/src/Adapter/Presenter/Product/ProductPresenter.php#L113-L115